File: board_dsl/src/main/java/com/dooho/board/repository/LikyRepository.java
package com.dooho.board.repository;

import com.dooho.board.entity.BoardEntity;
import com.dooho.board.entity.LikyEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Repository
@Transactional
public interface LikyRepository extends JpaRepository<LikyEntity,Integer> {

    Integer countByBoardNumber(Integer boardNumber);

    List<LikyEntity> findByBoardNumber(Integer boardNumber);

    void deleteByLikeUserNickname(String likeUserNickname);

    void deleteByLikeId(Integer likeId);

    void deleteByBoardNumberAndLikeUserNickname(Integer boardNumber, String likeUserNickname);
}

File: board_spring/src/main/java/com/dooho/board/repository/LikyRepository.kt
package com.dooho.board.repository

import com.dooho.board.entity.LikyEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional

@Repository
@Transactional
interface LikyRepository : JpaRepository<LikyEntity?, Int?> {
    fun countByBoardNumber(boardNumber: Int?): Int?
    fun findByBoardNumber(boardNumber: Int?): List<LikyEntity?>?
    fun deleteByLikeUserNickname(likeUserNickname: String?)
    fun deleteByLikeId(likeId: Int?)
    fun deleteByBoardNumberAndLikeUserNickname(boardNumber: Int?, likeUserNickname: String?)
}